src/libsystemd/sd-journal/journal-file.c

```diff
@@ -1630,24 +1670,27 @@ static int journal_file_append_field(
 }
 
 static Compression maybe_compress_payload(JournalFile *f, uint8_t *dst, const uint8_t *src, uint64_t size, size_t *rsize) {
-        Compression compression = COMPRESSION_NONE;
-
         assert(f);
         assert(f->header);
 
 #if HAVE_COMPRESSION
-        if (JOURNAL_FILE_COMPRESS(f) && size >= f->compress_threshold_bytes) {
-                compression = compress_blob(src, size, dst, size - 1, rsize);
-                if (compression > 0)
-                        log_debug("Compressed data object %"PRIu64" -> %zu using %s",
-                                  size, *rsize, compression_to_string(compression));
-                else
-                        /* Compression didn't work, we don't really care why, let's continue without compression */
-                        compression = COMPRESSION_NONE;
-        }
-#endif
+        Compression c = JOURNAL_FILE_COMPRESSION(f);
+
+        if (c == COMPRESSION_NONE || size < f->compress_threshold_bytes)
+                return COMPRESSION_NONE;
+
+        c = compress_blob_explicit(c, src, size, dst, size - 1, rsize);
+        if (c < 0 || c == COMPRESSION_NONE)
+                /* Compression didn't work, we don't really care why, let's continue without compression */
```

Comment:
too lazy to check, but I hope compress_blob_explicit() already debug logs about the reasons it might have failed?